crates/revm/src/context/inner_evm_context.rs

```diff
@@ -183,7 +184,13 @@ impl<DB: Database> InnerEvmContext<DB> {
         let code = if code.is_eof() {
             EOF_MAGIC_BYTES.clone()
         } else if code.is_eip7702() {
-            EIP7702_MAGIC_BYTES.clone()
+            // Reserve 23 bytes for EIP7702_MAGIC_BYTES (2 bytes) + EIP7702_VERSION (1 byte)
+            // + address (20 bytes)
+            let mut bytes = Vec::with_capacity(23);
+            bytes.extend_from_slice(&EIP7702_MAGIC_BYTES);
+            bytes.push(EIP7702_VERSION);
+            bytes.extend_from_slice(address.as_slice());
+            bytes.into()
```

Comment:
Hey, there is no need to craft the output, You need to use a function from the `code` iirc `code.original_bytes` should work.